solver/src/backends.rs

```diff
@@ -69,6 +69,16 @@ impl GenericBackend {
         self.opts.seed = seed;
         return self;
     }
+
+    /// Get the solver type.
+    pub fn get_solver_type(&self) -> SolverType {
+        self.solver_type
+    }
+
+    /// Get the solver timeout.
+    pub fn get_timeout_ms(&self) -> Option<usize> {
```

Comment:
You can name this `fn timeout_ms`

Reply:
This is the only comment I couldn't address because there's already such a function which sets the timeout value.